lymph/midline.py

```diff
@@ -758,17 +502,70 @@ def likelihood(
         except ValueError:
             return -np.inf if log else 0.
 
-        if len(self.patient_data.loc[(self.patient_data['info', 'tumor', 'midline_extension']==False) | (self.patient_data['info', 'tumor', 'midline_extension']==True)])>0:
+        llh = 0. if log else 1.
+
+        stored_t_stages = set(self.ext.ipsi.diagnose_matrices.keys())
+        provided_t_stages = set(self.ext.ipsi.diag_time_dists.keys())
+        t_stages = list(stored_t_stages.intersection(provided_t_stages))
+        
+        max_t = self.diag_time_dists.max_t
+        llh = 0. if log else 1.
+
+        state_probs_ipsi_nox = self.noext.ipsi._evolve_onestep()
+        state_probs_contra_nox = self.noext.contra._evolve_onestep()
+        state_probs_midext = self._evolve_midext(new_params = given_params)
+        state_probs_ipsi_ex = np.zeros(
+        shape=(max_t + 1, len(self.ext.ipsi.state_list)),
+        dtype=float
+        )
+        state_probs_ipsi_ex[0,0] = 1.
+        state_probs_contra_ex = np.zeros(
+        shape=(max_t + 1, len(self.ext.ipsi.state_list)),
+        dtype=float
+        )
+        state_probs_contra_ex[0,0] = 1.
+
+        for stage in t_stages:
+            for i in range(max_t):
+                state_probs_ipsi_ex[(i+1),:] = state_probs_midext[i,1] * state_probs_ipsi_ex[i,:] @ self.ext.ipsi.transition_matrix + state_probs_midext[i,0] * state_probs_ipsi_nox[(i+1),:]
```

Comment:
Isn't this line the actual one-step evolution?

Reply:
Yes indeed. I directly implemented it in the likelihood function as it needs to be calculated step by step by the previously computed states.

Reply:
Makes sense 👍🏻